firebase-functions/src/main/java/com/google/firebase/functions/PublisherStream.kt

```diff
@@ -296,9 +307,11 @@ internal class PublisherStream(
   private fun validateResponse(response: Response) {
     if (response.isSuccessful) return
 
-    val htmlContentType = "text/html; charset=utf-8"
     val errorMessage: String
-    if (response.code() == 404 && response.header("Content-Type") == htmlContentType) {
+    if (
+      response.code() == 404 &&
+        MediaType.get(response.header("Content-Type") ?: "").subtype() == "html"
```

Comment:
Should it be `?: ""`?

Reply:
Not sure if that's your question, but it should be `parse` instead of `get`. `parse` returns null and can be safely checked against "html" without throwing an exception.